autogpt_platform/backend/schema.prisma

```diff
@@ -20,55 +20,174 @@ model User {
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  userId      String
+  User        User          @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userGroupId String
+  UserGroup   UserGroup     @relation(fields: [userGroupId], references: [id], onDelete: Cascade)
+  Role        UserGroupRole @default(MEMBER)
+
+  @@unique([userId, userGroupId])
+  @@index([userId])
+  @@index([userGroupId])
 }
 
 // This model describes the Agent Graph/Flow (Multi Agent System).
-model AgentGraph {
+model Agent {
   id        String    @default(uuid())
   version   Int       @default(1)
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
   name        String?
   description String?
-  isActive    Boolean @default(true)
-  isTemplate  Boolean @default(false)
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  createdByUserId String
+  // Do not cascade delete the agent when the user is deleted
+  // This allows us to delete user data with deleting the agent which maybe in use by other users
+  CreatedByUser   User   @relation(fields: [createdByUserId], references: [id])
 
-  AgentNodes                  AgentNode[]
-  AgentGraphExecution         AgentGraphExecution[]
-  AgentGraphExecutionSchedule AgentGraphExecutionSchedule[]
+  groupId String?
+  // Do not cascade delete the agent when the group is deleted
+  // This allows us to delete user group data with deleting the agent which maybe in use by other users
+  Group   UserGroup? @relation(fields: [groupId], references: [id])
+
+  AgentNodes     AgentNode[]
+  AgentExecution AgentExecution[]
 
   // All sub-graphs are defined within this 1-level depth list (even if it's a nested graph).
-  AgentSubGraphs     AgentGraph[] @relation("AgentSubGraph")
-  agentGraphParentId String?
-  AgentGraphParent   AgentGraph?  @relation("AgentSubGraph", fields: [agentGraphParentId, version], references: [id, version], onDelete: Cascade)
+  SubAgents          Agent[] @relation("SubAgents")
+  agentParentId      String?
+  agentParentVersion Int?
+  AgentParent        Agent?  @relation("SubAgents", fields: [agentParentId, agentParentVersion], references: [id, version])
+
+  AgentPresets           AgentPreset[]
+  WebhookTrigger         WebhookTrigger[]
+  AgentExecutionSchedule AgentExecutionSchedule[]
+  UserAgents             UserAgent[]
+  UserBlockCredit        UserBlockCredit[]
+  StoreListing           StoreListing[]
+  StoreListingVersion    StoreListingVersion[]
+
+  @@id(name: "agentVersionId", [id, version])
+}
+
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+////////////////   USER SPECIFIC DATA   ////////////////////
+//////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////// 
+
+model AgentPreset {
```

Comment:
I think documenting what each of these tables is supposed to be for would be helpful long term

// Agent Presets is used to store????????? about an agent and is used